storage-node/src/services/webApi/app.ts

```diff
@@ -73,6 +74,16 @@ export async function createApp(config: AppConfig): Promise<Express> {
       }
     },
 
+    // Pre validate data object id in GET/HEAD file requests
+    (req: express.Request, res: express.Response<unknown, AppConfig>, next: NextFunction) => {
+      if (req.path === '/api/v1/files' && (req.method === 'GET' || req.method === 'HEAD')) {
+        validateDataObjectId(req, res)
+          .then(next)
+          .catch((error) => sendResponseWithError(res, next, error, 'upload'))
+      } else {
+        next()
+      }
+    },
```

Comment:
Good catch. Yeah I was surprised actually that the schema was for a string. Maybe it could be of type `number` rather than regex pattern on a string? Was a string perhaps chosen to support data object ids larger than javascript max number ?

Reply:
Even that regex expression could allow numbers too large for BN to fail.
With this custom function we could also do checks like if the id outside the range of valid dataobject ids?